authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationPlugin.java

```diff
@@ -0,0 +1,583 @@
+                .withName(newOwner.name())
+                .withRoleNames(Collections.emptyList())
+                .withRoleIds(Collections.emptyList())
+                .withAuditInfo(auditInfo)
+                .build();
+        onUserAdded(userEntity);
+      } else {
+        GroupEntity groupEntity =
+            GroupEntity.builder()
+                .withId(1L)
+                .withName(newOwner.name())
+                .withRoleNames(Collections.emptyList())
+                .withRoleIds(Collections.emptyList())
+                .withAuditInfo(auditInfo)
+                .build();
+        onGroupAdded(groupEntity);
+      }
+    }
+
+    RangerPolicy policy = rangerHelper.findManagedPolicy(metadataObject);
+    try {
+      if (policy == null) {
+        policy = rangerHelper.addOwnerToNewPolicy(metadataObject, newOwner);
+        rangerClient.createPolicy(policy);
+      } else {
+        rangerHelper.updatePolicyOwner(policy, preOwner, newOwner);
+        rangerClient.updatePolicy(policy.getId(), policy);
+      }
+    } catch (RangerServiceException e) {
+      throw new RuntimeException(e);
+    }
+
+    return Boolean.TRUE;
+  }
+
+  /**
+   * Grant the roles to the user. <br>
+   * 1. Create a user in the Ranger if the user does not exist. <br>
+   * 2. Create a role in the Ranger if the role does not exist. <br>
+   * 3. Add this user to the role. <br>
+   *
+   * @param roles The roles to grant to the group.
+   * @param user The user to grant the roles.
+   */
+  @Override
+  public Boolean onGrantedRolesToUser(List<Role> roles, User user) throws RuntimeException {
+    // If the user does not exist, then create it.
+    onUserAdded(user);
+
+    roles.stream()
+        .forEach(
+            role -> {
+              rangerHelper.createRangerRoleIfNotExists(role.name());
+              GrantRevokeRoleRequest grantRevokeRoleRequest =
+                  rangerHelper.createGrantRevokeRoleRequest(role.name(), user.name(), null);
+              try {
+                rangerClient.grantRole(rangerServiceName, grantRevokeRoleRequest);
+              } catch (RangerServiceException e) {
+                // ignore exception, support idempotent operation
+                LOG.warn("Grant role: {} to user: {} failed!", role, user, e);
+              }
+            });
+
+    return Boolean.TRUE;
+  }
+
+  /**
+   * Revoke the roles from the user. <br>
+   * 1. Create a user in the Ranger if the user does not exist. <br>
+   * 2. Check the role if it exists in the Ranger. <br>
+   * 3. Revoke this user from the role. <br>
+   *
+   * @param roles The roles to revoke the group.
+   * @param user The user to revoke the roles.
+   */
+  @Override
+  public Boolean onRevokedRolesFromUser(List<Role> roles, User user) throws RuntimeException {
+    // If the user does not exist, then create it.
+    onUserAdded(user);
+
+    roles.stream()
+        .forEach(
+            role -> {
+              rangerHelper.checkRangerRole(role.name());
```

Comment:
As you have ignored the result of `checkRangerRole`, how can you assure that the role exists?

Reply:
If this Role doesn't exist the `checkRangerRole()` will throw exception to terminate the operation.

Reply:
<img width="1064" alt="image" src="https://github.com/user-attachments/assets/c33f1a5f-eac7-4a9c-897c-9e26b91ea5d3">

So, when will the role be null in the method above?

Reply:
1. I want to know if the value of the role is null means the role does not exist or something else.
2. If the role could not be null, so I don't think we should keep the logic.

Reply:
If we return the value, we should check the return value. It's more safer. Otherwise, it will be weird for other maintainers.